tests/network/bindingplugin_macvtap.go

```diff
@@ -0,0 +1,225 @@
+		})
+
+		Context("with live traffic", func() {
+			var serverVMI *v1.VirtualMachineInstance
+			var serverVMIPodName string
+			var serverIP string
+
+			const macvtapIfaceIPReportTimeout = 4 * time.Minute
+
+			BeforeEach(func() {
+				serverVMI = startVMI(libvmi.NewAlpineWithTestTooling(
+					libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+					libvmi.WithInterface(*libvmi.InterfaceWithMac(
+						libvmi.InterfaceWithMacvtapBindingPlugin(macvtapNetworkName), serverMAC)),
+					libvmi.WithNetwork(v1.DefaultPodNetwork()),
+					libvmi.WithNetwork(libvmi.MultusNetwork(macvtapNetworkName, macvtapNetworkName)),
+				), console.LoginToAlpine)
+
+				Expect(serverVMI.Status.Interfaces).NotTo(BeEmpty(), "a migrate-able VMI must have network interfaces")
+				serverVMIPodName = tests.GetVmPodName(kubevirt.Client(), serverVMI)
+				var err error
+				serverIP, err = waitVMMacvtapIfaceIPReport(serverVMI, serverMAC, macvtapIfaceIPReportTimeout)
+				Expect(err).NotTo(HaveOccurred(), "should have managed to figure out the IP of the server VMI")
+			})
+
+			BeforeEach(func() {
+				// TODO test also the IPv6 address (issue- https://github.com/kubevirt/kubevirt/issues/7506)
+				libnet.SkipWhenClusterNotSupportIpv4()
+				Expect(libnet.PingFromVMConsole(clientVMI, serverIP)).To(Succeed(), "connectivity is expected *before* migrating the VMI")
+			})
+
+			It("should keep connectivity after a migration", func() {
+				const containerCompletionWaitTime = 60
+				migration := libmigration.New(serverVMI.Name, serverVMI.GetNamespace())
+				_ = libmigration.RunMigrationAndExpectToCompleteWithDefaultTimeout(kubevirt.Client(), migration)
+				// In case of clientVMI and serverVMI running on the same node before migration, the serverVMI
+				// will be reachable only when the original launcher pod terminates.
+				Eventually(func() error {
+					return waitForPodCompleted(serverVMI.Namespace, serverVMIPodName)
+				}, containerCompletionWaitTime, time.Second).Should(Succeed(), fmt.Sprintf("all containers should complete in source virt-launcher pod: %s", serverVMIPodName))
+				Expect(libnet.PingFromVMConsole(clientVMI, serverIP)).To(Succeed(), "connectivity is expected *after* migrating the VMI")
+			})
+		})
+	})
+})
+
+func startVMI(vmi *v1.VirtualMachineInstance, loginFn console.LoginToFunction) *v1.VirtualMachineInstance {
+	vmi, err := kubevirt.Client().VirtualMachineInstance(testsuite.GetTestNamespace(nil)).Create(context.Background(), vmi)
+	Expect(err).ToNot(HaveOccurred(), "should create VMI successfully")
+	return libwait.WaitUntilVMIReady(vmi, loginFn)
+}
+
+func waitForPodCompleted(podNamespace string, podName string) error {
+	pod, err := kubevirt.Client().CoreV1().Pods(podNamespace).Get(context.TODO(), podName, k8smetav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	if pod.Status.Phase == k8sv1.PodSucceeded || pod.Status.Phase == k8sv1.PodFailed {
```

Comment:
I would remove the `pod.Status.Phase == k8sv1.PodFailed`. It doesn't make sense, I don't know why it was there in the first place.

Reply:
I think it is correct.
The source pod can either terminate gracefully or in a fail state, it does not matter. The important part is that it will terminate.

Ref: https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase